com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java

```diff
@@ -167,30 +168,44 @@ public void setHitCount(int hitCount) {
         return locations;
     }
 
-    private static List<Location> collectLocations(List<ReferenceType> refTypes, int lineNumber) {
+    private static List<Location> collectLocations(List<ReferenceType> refTypes, int lineNumber, boolean includeNestedTypes) {
         List<Location> locations = new ArrayList<>();
         try {
             refTypes.forEach(refType -> {
-                locations.addAll(collectLocations(refType, lineNumber));
-                locations.addAll(collectLocations(refType.nestedTypes(), lineNumber));
+                List<Location> newLocations = collectLocations(refType, lineNumber);
+                if (!newLocations.isEmpty()) {
+                    locations.addAll(newLocations);
+                } else if (includeNestedTypes) {
```

Comment:
just if?

Reply:
what's the advice?

Reply:
i mean why here it is `else if` rather than `if`?

Reply:
if the breakpoint located in the refType, no need to search the nested type.